src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ShardStateCollector.java

```diff
@@ -0,0 +1,139 @@
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics.SHARD_PRIMARY;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics.SHARD_REPLICA;
+
+public class ShardStateCollector extends PerformanceAnalyzerMetricsCollector implements MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL = MetricsConfiguration.CONFIG_MAP.get(ShardStateCollector.class).samplingInterval;
+    private static final Logger LOG = LogManager.getLogger(ShardStateCollector.class);
+    private static final int KEYS_PATH_LENGTH = 0;
+    private StringBuilder value;
+
+    public ShardStateCollector() {
+        super(SAMPLING_TIME_INTERVAL, "ShardsStateCollector");
+        value = new StringBuilder();
+    }
+
+    @Override
+    void collectMetrics( long startTime) {
+        long mCurrT = System.currentTimeMillis();
+        if (ESResources.INSTANCE.getClusterService() == null) {
+            return;
+        }
+        ClusterState clusterState = ESResources.INSTANCE.getClusterService().state();
+        boolean inActiveShard = false;
+        try {
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+            .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            RoutingTable routingTable = clusterState.routingTable();
+            String[] indices = routingTable.indicesRouting().keys().toArray(String.class);
+            for (String index : indices) {
+                List<ShardRouting> allShardsIndex = routingTable.allShards(index);
+                value.append(createJsonObject(AllMetrics.ShardStateDimension.INDEX_NAME.toString(), index));
+                for (ShardRouting shard :allShardsIndex) {
+                    String nodeName = StringUtils.EMPTY;
+                    if (shard.assignedToNode()) {
+                        nodeName = clusterState.nodes().get(shard.currentNodeId()).getName();
+                    }
+                    if (shard.state() != ShardRoutingState.STARTED) {
+                        inActiveShard = true;
+                        value
+                                .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor)
+                                .append(new ShardStateMetrics(
+                                        shard.getId(),
+                                        shard.primary() ? SHARD_PRIMARY : SHARD_REPLICA,
+                                        nodeName,
+                                        shard.state().name())
+                                        .serialize());
+
+                    }
+                }
+            }
+            value.append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            if(inActiveShard) {
+                saveMetricValues(value.toString(), startTime);
+            }
+            PerformanceAnalyzerApp.ERRORS_AND_EXCEPTIONS_AGGREGATOR.updateStat(
```

Comment:
This is not error, so should not be part of `ERRORS_AND_EXCEPTIONS_AGGREGATOR`

Reply:
You can refer to [this PR](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/462/files).